android/src/main/java/com/audioStreaming/Signal.java

```diff
@@ -300,24 +312,48 @@ public void playerPCMFeedBuffer(boolean isPlaying, int bufSizeMs, int bufCapacit
             this.isPreparingStarted = false;
             if (bufSizeMs < 500) {
                 this.isPlaying = false;
-                sendBroadcast(new Intent(Mode.BUFFERING_START));
+                Intent BroadcastIntent = new Intent(Mode.BUFFERING_START);
+                BroadcastIntent.putExtra("url", this.streamingURL);
+                this.duration = bufCapacityMs / 1000.0;
+                this.progress = (bufCapacityMs - bufSizeMs) / 1000.0;
+                BroadcastIntent.putExtra("duration", this.duration);
+                BroadcastIntent.putExtra("progress", this.progress);
+                sendBroadcast(BroadcastIntent);
                 //buffering
             } else {
                 this.isPlaying = true;
-                sendBroadcast(new Intent(Mode.PLAYING));
+                Intent BroadcastIntent = new Intent(Mode.PLAYING);
+                BroadcastIntent.putExtra("url", this.streamingURL);
+                this.duration = bufCapacityMs / 1000.0;
+                this.progress = (bufCapacityMs - bufSizeMs) / 1000.0;
+                BroadcastIntent.putExtra("duration", this.duration);
+                BroadcastIntent.putExtra("progress", this.progress);
+
+                sendBroadcast(BroadcastIntent);
                 //playing
             }
         } else {
             //buffering
             this.isPlaying = false;
-            sendBroadcast(new Intent(Mode.BUFFERING_START));
+            Intent BroadcastIntent = new Intent(Mode.BUFFERING_START);
+            BroadcastIntent.putExtra("url", this.streamingURL);
+            this.duration = bufCapacityMs / 1000.0;
+            this.progress = (bufCapacityMs - bufSizeMs) / 1000.0;
+            BroadcastIntent.putExtra("duration", this.duration);
+            BroadcastIntent.putExtra("progress", this.progress);
+            sendBroadcast(BroadcastIntent);
         }
     }
 
     @Override
     public void playerException(final Throwable t) {
         this.isPlaying = false;
         this.isPreparingStarted = false;
+        StringWriter sw = new StringWriter();
+        PrintWriter pw = new PrintWriter(sw);
+        t.printStackTrace(pw);
+        Intent BroadcastIntent = new Intent(Mode.ERROR);
+        BroadcastIntent.putExtra("stacktrace", sw.toString());
```

Comment:
Forgot to remove dead code. 

Nowadays, when there is an error that is hard to see why (some of my file were private on AWS and I took me some time). But I agree, it is another feature.

I'm gonna update this PR.